File: Single-Linked-List/insert_node_at_end.py
# File Name: insert_at_end.py

# Node class
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

# Singly Linked List class
class SinglyLinkedList:
    def __init__(self):
        self.head = None

    # Method to insert a node at the end
    def insert_end(self):
        value = int(input("Enter data for new node: "))
        new_node = Node(value)

        # If list is empty
        if self.head is None:
            self.head = new_node
        else:
            temp = self.head
            while temp.next:
                temp = temp.next
            temp.next = new_node

File: Single-Linked-List/InsertNodeAtEnd.java
/* File Name: InsertAtEnd.java */

import java.util.Scanner;

// Node class
class Node {
    int data;
    Node next;

    Node(int data) {
        this.data = data;
        this.next = null;
    }
}

// Singly Linked List class
public class InsertAtEnd {

    Node head = null;

    // Method to insert a node at the end
    void insertEnd() {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter data for new node: ");
        int value = sc.nextInt();

        Node newNode = new Node(value);

        // If list is empty
        if (head == null) {
            head = newNode;
        } else {
            Node temp = head;
            while (temp.next != null)
                temp = temp.next;
            temp.next = newNode;
        }
    }
}
